backend/src/main/java/com/bakdata/conquery/io/result/arrow/ArrowRenderer.java

```diff
@@ -221,23 +221,28 @@ private static RowConsumer structVectorFiller(StructVector vector, RowConsumer [
             vector.setIndexDefined(rowNumber);
         };
     }
-    private static RowConsumer listVectorFiller(ListVector vector, RowConsumer nestedConsumer, Function<Object[], Object[]> resultExtractor){
+    private static RowConsumer listVectorFiller(ListVector vector, RowConsumer nestedConsumer, Function<Object[], List> resultExtractor){
         // This is not used at the moment see ResultType.ListT::getArrowFieldType
         return (rowNumber, line) -> {
             // Values is a vertical list
-            Object[] values = resultExtractor.apply(line);
+            List values = resultExtractor.apply(line);
             if (values == null) {
                 vector.setNull(rowNumber);
                 return;
             }
 
-            vector.startNewValue(rowNumber);
-            for (int i = 0; i < values.length; i++) {
+            int start = vector.startNewValue(rowNumber);
+            for (int i = 0; i < values.size(); i++) {
                 // These short lived one value arrays are a workaround at the moment
-                nestedConsumer.accept(i, new Object[] {values[i]});
+                nestedConsumer.accept(start + i, new Object[] {values.get(i)});
             }
 
-            vector.endValue(rowNumber,values.length);
+            // Workaround for https://issues.apache.org/jira/browse/ARROW-8842
+            final FieldVector innerVector = vector.getDataVector();
+            int valueCount = innerVector.getValueCount();
```

Comment:
Bist du dir sicher? Ist das hier nicht einfach 0?

Reply:
Ich erweitere mal den Teste um noch mehr werte in dem ListVector

Reply:
So ist der Fix richtig, Ich habe die Batchsize mal erhöht und mehr werte in die Spalte gepackt. Wenn man hier immer von  0 anfängt, dann über schreibt man frühere Werte in dem selben Batch